base/abstractarraymath.jl

```diff
@@ -377,7 +377,8 @@ _reperr(s, n, N) = throw(ArgumentError("number of " * s * " repetitions " *
 
     # fill the first inner block
     if all(x -> x == 1, inner)
-        R[axes(A)...] = A
+        idxs = (axes(A)..., ntuple(n->OneTo(1), length(shape) - length(inner))...) # keep dimension consistent
```

Comment:
Rather than `OneTo(1)`, it might be slightly more friendly to bizarro array subtypes to just use `:`s here.

Reply:
I was trying to figure out a simple way of getindex. Got it!